internal/app/machined/pkg/system/services/etcd.go

```diff
@@ -359,3 +317,85 @@ func buildInitialCluster(config runtime.Configurator, name, ip string) (initial
 
 	return initial, nil
 }
+
+func blacklistError(s string) error {
+	return fmt.Errorf("extra etcd arg %q is not allowed", s)
+}
+
+// nolint: gocyclo
+func args(config runtime.Configurator) ([]string, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := metadata.Open()
+	if err != nil {
+		return nil, err
+	}
+
+	ips, err := net.IPAddrs()
+	if err != nil {
+		return nil, fmt.Errorf("failed to discover IP addresses: %w", err)
+	}
+
+	if len(ips) == 0 {
+		return nil, errors.New("failed to discover local IP")
+	}
+
+	a := argumentor.Args{
```

Comment:
could we rename `a` to `defaultArgs` or something like that?

Reply:
They're not default in the sense that they can be overwritten. Will come up with something more descriptive.